src/components/DepositWidget/index.tsx

```diff
@@ -83,6 +84,39 @@ const DepositWidget: React.FC = () => {
   const { userAddress } = useWalletConnection()
   const { balances, setBalances, error } = useTokenBalances()
 
+  const withdrawRequest = useRef({ amount: null, tokenAddress: null, pendingAmount: null, symbol: null })
```

Comment:
Why add indirect _state_ with `useRef`, just do `useState`, it'll update before Modal buttons are created anyway.

Reply:
My idea was to avoid triggering a redraw by using those values as `instance` variables. I'll try as state as you suggest.

Reply:
Updated.

Reply:
`toggleModal` does `setState` internally, so rerender is triggered either way. And as your `setWithdrawBalance` is close together with `toggleModal` and they are one sync operation (even if they are inside an async function, because await appears later), state is updated once only.
Or that's how I understand it works.